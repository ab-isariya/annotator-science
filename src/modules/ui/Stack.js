import PropTypes from 'prop-types';

/**
 * Simple component to stack elements on top of another. AS OF NOW, IT CENTERS
 */
const Stack = ({children, className, ...otherProps}) => {
  return (
    <div className={`${className ? className : ''} relative`} {...otherProps}>
      {children[0]}
      <div
        className="absolute top-2/4 left-2/4"
        style={{
          transform: 'translate(-50%, -50%)'
        }}>
        {children[1]}
      </div>
    </div>
  );
};

Stack.propTypes = {
  /** Any additional class for component */
  className: PropTypes.string,
  //React node children
  children: PropTypes.node.isRequired
};

export default Stack;
