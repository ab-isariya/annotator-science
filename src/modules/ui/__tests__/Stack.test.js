import Stack from '../Stack';

import renderer from 'react-test-renderer';

describe('<Stack/>', () => {
  it('Renders without crashing', () => {
    const comp = renderer
      .create(
        <Stack>
          <p>Hello</p> <p>World</p>
        </Stack>
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
  });

  it('Renders without crashing with only 1 child', () => {
    const comp = renderer
      .create(
        <Stack>
          <p>Hello</p>
        </Stack>
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.children.length).toBe(1);
  });

  it('Renders without crashing with more than 2 children', () => {
    const comp = renderer
      .create(
        <Stack>
          <p>Hello</p>
          <p>World</p>
          <p>How are you?</p>
        </Stack>
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.children.length).toBe(2);
    expect(comp.children[1].children.length).toBe(1);
  });

  it('Renders with a custom classname', () => {
    const comp = renderer
      .create(
        <Stack className="testClass">
          <p>Hello</p>
          <p>World</p>
        </Stack>
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/testClass/i);
  });
});
