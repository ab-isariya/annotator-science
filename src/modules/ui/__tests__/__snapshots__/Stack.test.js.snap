// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stack/> Renders with a custom classname 1`] = `
<div
  className="testClass relative"
>
  <p>
    Hello
  </p>
  <div
    className="absolute top-2/4 left-2/4"
    style={
      Object {
        "transform": "translate(-50%, -50%)",
      }
    }
  >
    <p>
      World
    </p>
  </div>
</div>
`;

exports[`<Stack/> Renders without crashing 1`] = `
<div
  className=" relative"
>
  <p>
    Hello
  </p>
  <div
    className="absolute top-2/4 left-2/4"
    style={
      Object {
        "transform": "translate(-50%, -50%)",
      }
    }
  >
     
  </div>
</div>
`;

exports[`<Stack/> Renders without crashing with more than 2 children 1`] = `
<div
  className=" relative"
>
  <p>
    Hello
  </p>
  <div
    className="absolute top-2/4 left-2/4"
    style={
      Object {
        "transform": "translate(-50%, -50%)",
      }
    }
  >
    <p>
      World
    </p>
  </div>
</div>
`;

exports[`<Stack/> Renders without crashing with only 1 child 1`] = `
<div
  className=" relative"
>
  <div
    className="absolute top-2/4 left-2/4"
    style={
      Object {
        "transform": "translate(-50%, -50%)",
      }
    }
  />
</div>
`;
